osf-packager/src/main/java/org/dataconservancy/cos/packaging/OsfContentProvider.java

```diff
@@ -337,44 +236,17 @@ private static FileInfo contentFromUrl(String filename, String contentUrl, Relat
     }
 
     /**
-     * Creates a temporary directory under {@code java.io.tmpdir}.  The directory will be uniquely named, so as to
-     * avoid any filename conflicts with simultaneously executing downloads (e.g. another packager running in another
-     * JVM).
-     *
-     * @return the temporary directory
-     * @throws IOException if the directory cannot be allocated
-     */
-    static File allocateTempDir() throws IOException {
```

Comment:
This proved to cause some problems with the new code layout.  The ContentProvider can't delete the temporary folder until the IpmPackager is done using it, so I created a close() method on the ContentProvide that has to be called by its owner when the IpmPackager is done.  I initially put that call into IpmPackager.buildPackage(), but since the packager doesn't own the provider that seemed like the wrong place.

Anyway, The OsfContentProviderTest tests run now, but there is an exception when they try to delete a file in the temporary folder.  When I run in the debugger, the problem doesn't happen, which is maddening.  I can also delete the file after the test ends.  Maybe the process that creates the file doesn't close its File object right after creation?  I will create an Issue for this, as well as another delete error I've seen since I started work on the wiki stuff - the package tar file cannot be deleted after the gz file is created from it.